x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/aggregation/blockhash/BlockHashTests.java

```diff
@@ -1232,6 +1235,190 @@ public void testLongNull() {
         }, blockFactory.newLongArrayVector(values, values.length).asBlock(), blockFactory.newConstantNullBlock(values.length));
     }
 
+    public void test2BytesRefsKeys() {
+        final Page page;
+        final int positions1 = 100_000;
+        final int positions2 = 10;
+        final int totalPositions = positions1 * positions2;
+        try (
+            BytesRefBlock.Builder builder1 = blockFactory.newBytesRefBlockBuilder(totalPositions);
+            BytesRefBlock.Builder builder2 = blockFactory.newBytesRefBlockBuilder(totalPositions);
+        ) {
+            for (int i = 0; i < positions1; i++) {
+                for (int p = 0; p < positions2; p++) {
+                    builder1.appendBytesRef(new BytesRef("abcdef" + i));
+                    builder2.appendBytesRef(new BytesRef("abcdef" + p));
+                }
+            }
+            page = new Page(builder1.build(), builder2.build());
+        }
```

Comment:
Do we want this all in one page? I know the `BlockHash` needs it in memory. And we build that `HashMap` with it in memory below. But maybe we make `Page`s of a reasonable size here? 

Meh. I guess it's only like 9mb. Probably not worth changing it unless you need to go bigger.